rs/types/management_canister_types/src/lib.rs

```
@@ -808,6 +834,106 @@ impl From<pb_canister_state_bits::LogVisibility> for LogVisibility {
+                Self::AllowedViewers(_) => Self::default(),
+                _ => record,
+            },
+            Err(_) => LogVisibility::decode(blob)?.into(),
+        };
+        Ok(args)
+    }
+}
+
+impl From<LogVisibility> for LogVisibilityV2 {
+    fn from(item: LogVisibility) -> Self {
+        match item {
+            LogVisibility::Controllers => Self::Controllers,
+            LogVisibility::Public => Self::Public,
+        }
+    }
+}
+
+impl From<&LogVisibilityV2> for pb_canister_state_bits::LogVisibilityV2 {
+    fn from(item: &LogVisibilityV2) -> Self {
+        use pb_canister_state_bits as pb;
+        match item {
+            LogVisibilityV2::Controllers => pb::LogVisibilityV2 {
+                log_visibility_v2: Some(pb::log_visibility_v2::LogVisibilityV2::Controllers(1)),
+            },
+            LogVisibilityV2::Public => pb::LogVisibilityV2 {
+                log_visibility_v2: Some(pb::log_visibility_v2::LogVisibilityV2::Public(2)),
+            },
+            LogVisibilityV2::AllowedViewers(principals) => pb::LogVisibilityV2 {
+                log_visibility_v2: Some(pb::log_visibility_v2::LogVisibilityV2::AllowedViewers(
+                    pb::LogVisibilityAllowedViewers {
+                        principals: principals
+                            .get()
+                            .iter()
+                            .map(|c| (*c).into())
+                            .collect::<Vec<ic_protobuf::types::v1::PrincipalId>>()
+                            .clone(),
+                    },
+                )),
+            },
+        }
+    }
+}
+
+impl From<pb_canister_state_bits::LogVisibilityV2> for LogVisibilityV2 {
+    fn from(item: pb_canister_state_bits::LogVisibilityV2) -> Self {
+        use pb_canister_state_bits as pb;
+        match item.log_visibility_v2 {
+            None => panic!("Invalid LogVisibility enum variant"),
```

I'd return an error instead as per other cases where conversion from proto to Rust might fail and handle it when we deserialize a checkpoint similar to other cases (handling can simply be propagating the error which would result in a panic eventually but it's less surprising I think since it's the usual pattern).

---

Done.